utoipa-swagger-ui/build.rs

```diff
@@ -22,34 +22,40 @@ use zip::{result::ZipError, ZipArchive};
 const SWAGGER_UI_DOWNLOAD_URL_DEFAULT: &str =
     "https://github.com/swagger-api/swagger-ui/archive/refs/tags/v5.17.3.zip";
 
+const SWAGGER_UI_DOWNLOAD_URL: &str = "SWAGGER_UI_DOWNLOAD_URL";
+const SWAGGER_UI_OVERWRITE_FOLDER: &str = "SWAGGER_UI_OVERWRITE_FOLDER";
+
 fn main() {
     let target_dir = env::var("OUT_DIR").unwrap();
-    println!("OUT_DIR: {}", target_dir);
+    println!("OUT_DIR: {target_dir}");
 
     let url =
-        env::var("SWAGGER_UI_DOWNLOAD_URL").unwrap_or(SWAGGER_UI_DOWNLOAD_URL_DEFAULT.to_string());
+        env::var(SWAGGER_UI_DOWNLOAD_URL).unwrap_or(SWAGGER_UI_DOWNLOAD_URL_DEFAULT.to_string());
 
-    println!("SWAGGER_UI_DOWNLOAD_URL: {}", url);
+    println!("{SWAGGER_UI_DOWNLOAD_URL}: {url}");
     let zip_filename = url.split('/').last().unwrap().to_string();
     let zip_path = [&target_dir, &zip_filename].iter().collect::<PathBuf>();
 
-    if !zip_path.exists() {
-        if url.starts_with("http://") || url.starts_with("https://") {
-            println!("start download to : {:?}", zip_path);
-            download_file(&url, zip_path.clone()).unwrap();
-        } else if url.starts_with("file://") {
-            let file_path = Url::parse(&url).unwrap().to_file_path().unwrap();
-            println!("start copy to : {:?}", zip_path);
-            fs::copy(file_path, zip_path.clone()).unwrap();
-        } else {
-            panic!("invalid SWAGGER_UI_DOWNLOAD_URL: {} -> must start with http:// | https:// | file://", url);
-        }
+    if url.starts_with("file:") {
+        let mut file_path = url::Url::parse(&url).unwrap().to_file_path().unwrap();
+        file_path = fs::canonicalize(file_path).expect("swagger ui download path should exists");
+
+        // with file protocol utoipa swagger ui should compile when file changes
+        println!("cargo:rerun-if-changed={:?}", file_path);
+
+        println!("start copy to : {:?}", zip_path);
+        fs::copy(file_path, zip_path.clone()).unwrap();
```

Comment:
At this point this _really_ just seems equivalent to `zip_path = file_path;`.

Reply:
Yeah, while this is little bit older commit but in practice yes, the copy seems to be superfluous since it is anyway extracted to the `OUTPUT` dir. But perhaps we can live with that :laughing: